RabbitMQ.Stream.Client/Producer.cs

```diff
@@ -188,18 +188,35 @@ async Task SendMessages(List<(ulong, Message)> messages)
             }
         }
 
-        public async Task<ResponseCode> Close()
+        public Task<ResponseCode> Close()
         {
             if (client.IsClosed)
             {
-                return ResponseCode.Ok;
+                return Task.FromResult(ResponseCode.Ok);
+            }
+
+            var result = ResponseCode.Ok;
+            try
+            {
+                var deletePublisherResponseTask = client.DeletePublisher(publisherId);
```

Comment:
Agree, but in this specific case, I need to avoid waiting 10 seconds for the close. As you can see:
```csharp
 deletePublisherResponseTask.Wait(TimeSpan.FromSeconds(3));
```
The call is still Async but just wait for 3 seconds. Note that this call is only when the producer is closing. 
No impact on the performances in general.

Reply:
OK. I'll open an issue to look a little more into using `Wait` instead of `await`. Since this isn't a hot path it's not as important.

Reply:
https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/issues/106